src/governance.cpp

```diff
@@ -1014,6 +1023,123 @@ bool CGovernanceObject::IsValidLocally(const CBlockIndex* pindex, std::string& s
+        return false;
+    }
+    
+    uint256 nExpectedHash = GetHash();
+
+    CTransaction txCollateral;
+    uint256 nBlockHash;
+
+    // RETRIEVE TRANSACTION IN QUESTION  
+
+    if(!GetTransaction(nCollateralHash, txCollateral, Params().GetConsensus(), nBlockHash, true)){
+        strError = strprintf("Can't find collateral tx %s", txCollateral.ToString());
+        LogPrintf("CGovernanceObject::IsCollateralValid - %s\n", strError);
+        return false;
+    }
+
+    if(txCollateral.vout.size() < 1) {
+        strError = strprintf("tx vout size less than 1 | %d", txCollateral.vout.size());
+        LogPrintf ("CGovernanceObject::IsCollateralValid - %s\n", strError);
+        return false;
+    }
+
+    // LOOK FOR SPECIALIZED GOVERNANCE SCRIPT (PROOF OF BURN)
+
+    CScript findScript;
+    findScript << OP_RETURN << ToByteVector(nExpectedHash);
+
+    DBG( cout << "IsCollateralValid txCollateral.vout.size() = " << txCollateral.vout.size() << endl; );
+
+    DBG( cout << "IsCollateralValid: findScript = " << ScriptToAsmStr( findScript, false ) << endl; );
+
+    DBG( cout << "IsCollateralValid: nMinFee = " << nMinFee << endl; );
+
+
+    bool foundOpReturn = false;
+    BOOST_FOREACH(const CTxOut o, txCollateral.vout) {
+        DBG( cout << "IsCollateralValid txout : " << o.ToString() 
+             << ", o.nValue = " << o.nValue
+             << ", o.scriptPubKey = " << ScriptToAsmStr( o.scriptPubKey, false )
+             << endl; );
+        if(!o.scriptPubKey.IsNormalPaymentScript() && !o.scriptPubKey.IsUnspendable()){
+            strError = strprintf("Invalid Script %s", txCollateral.ToString());
+            LogPrintf ("CGovernanceObject::IsCollateralValid - %s\n", strError);
+            return false;
+        }
+        if(o.scriptPubKey == findScript && o.nValue >= nMinFee) {
+            DBG( cout << "IsCollateralValid foundOpReturn = true" << endl; );
+            foundOpReturn = true;
+        }
+        else  {
+            DBG( cout << "IsCollateralValid No match, continuing" << endl; );
+        }
+
+    }
+
+    if(!foundOpReturn){
+        strError = strprintf("Couldn't find opReturn %s in %s", nExpectedHash.ToString(), txCollateral.ToString());
+        LogPrintf ("CGovernanceObject::IsCollateralValid - %s\n", strError);
+        return false;
+    }
+
+    // GET CONFIRMATIONS FOR TRANSACTION
+
+    LOCK(cs_main);
+    int nConfirmationsIn = GetIXConfirmations(nCollateralHash);
+    if (nBlockHash != uint256()) {
+        BlockMap::iterator mi = mapBlockIndex.find(nBlockHash);
+        if (mi != mapBlockIndex.end() && (*mi).second) {
+            CBlockIndex* pindex = (*mi).second;
+            if (chainActive.Contains(pindex)) {
+                nConfirmationsIn += chainActive.Height() - pindex->nHeight + 1;
+            }
+        }
+    }
+
+    //if we're syncing we won't have instantX information, so accept 1 confirmation 
+    if(nConfirmationsIn >= GOVERNANCE_FEE_CONFIRMATIONS) {
+        strError = "valid";
```

Comment:
I would avoid parsing pure strings for errors tbh. Should use some error id instead in this case imo. Right now the only reason I see for string like this is https://github.com/dashpay/dash/pull/988#discussion_r77445257 which already is not too nice. If still some parsing loginc is involved I would rather see "error" == "" as "valid" response. If function returned false and strError is "" (like in the line I linked to) that must be a clear red flag that some logic branch is missing error description. Also I don't see any rpc passing this string in cases when function returned true.